fuse_models/include/fuse_models/unicycle_2d_predict.h

```diff
@@ -100,6 +101,105 @@ inline void predict(
   fuse_core::wrapAngle2D(yaw2);
 }
 
+enum StateIndex : uint8_t
+{
+  X,
+  Y,
+  YAW,
+  V_X,
+  V_Y,
+  V_YAW,
+  A_X,
+  A_Y
+};
+
+/**
+ * @brief Given a state and time delta, predicts a new state
+ * @param[in] position1_x - First X position
+ * @param[in] position1_y - First Y position
+ * @param[in] yaw1 - First orientation
+ * @param[in] vel_linear1_x - First X velocity
+ * @param[in] vel_linear1_y - First Y velocity
+ * @param[in] vel_yaw1 - First yaw velocity
+ * @param[in] acc_linear1_x - First X acceleration
+ * @param[in] acc_linear1_y - First Y acceleration
+ * @param[in] dt - The time delta across which to predict the state
+ * @param[in] position2_x - Second X position
+ * @param[in] position2_y - Second Y position
+ * @param[in] yaw2 - Second orientation
+ * @param[in] vel_linear2_x - Second X velocity
+ * @param[in] vel_linear2_y - Second Y velocity
+ * @param[in] vel_yaw2 - Second yaw velocity
+ * @param[in] acc_linear2_x - Second X acceleration
+ * @param[in] acc_linear2_y - Second Y acceleration
+ * @param[in] jacobians - Jacobians wrt the state
+ */
+// TODO(efernandez) fuse this with the templated predict function so we can use the analytic Jacobian in the cost
+// function. See http://ceres-solver.org/analytical_derivatives.html
+inline void predict(
+  const double position1_x,
+  const double position1_y,
+  const double yaw1,
+  const double vel_linear1_x,
+  const double vel_linear1_y,
+  const double vel_yaw1,
+  const double acc_linear1_x,
+  const double acc_linear1_y,
+  const double dt,
+  double& position2_x,
+  double& position2_y,
+  double& yaw2,
+  double& vel_linear2_x,
+  double& vel_linear2_y,
+  double& vel_yaw2,
+  double& acc_linear2_x,
+  double& acc_linear2_y,
+  // TODO(efernandez) double** jacobians didn't play well with Eigen::Map<>, at least directly
+  double* jacobians)
+{
+  const double sy = ceres::sin(yaw1);
```

Comment:
+1 for merging the methods, but if we don't, then there's no reason to use the `ceres` versions of these, right?

Reply:
Will try merging the methods.